doc/SONiC_YANG_Model_Guidelines.md

```diff
@@ -6,46 +6,153 @@
+* As *CVL YANG* for configuration validation only, when standard YANG model (Openconfig, IETF etc.) is used as Northbound Management YANG. In such case steps #23 to #25 below can be skipped while developing SONiC YANG model.
 
 
 ## Guidelines
 
-### 1. Each primary section of ABNF.json (i.e a dictionary in ABNF.json) for Example, VLAN, VLAN_MEMBER, INTERFACE  in ABNF.json will be mapped to a container in YANG model.
 
-Example: Table INTERFACE will translate to container INTERFACE.
+### 1. All schema definitions related to a feature should be written in a single YANG model file. YANG model file is named as 'sonic-{feature}.yang', e.g. for ACL feature sonic-acl.yang is the file name. It is mandatory to define a top level YANG container named as 'sonic-{feature}' i.e. same as YANG model name. For example, sonic-acl.yang should have 'sonic-acl' as top level container. All other definition should be written inside the top level container.
+
+Example :
+####  YANG
+```
+module sonic-acl {
+	container sonic-acl {
+		.....
+		.....
+	}
+}
+```
+
+
+### 2  It is best to categorize YANG Modules based on a networking components. For example, it is good to have separate modules for VLAN,  ACL, PORT and IP-ADDRESSES etc.
+```
+sonic-acl.yang
+sonic-interface.yang
+sonic-port.yang
+sonic-vlan.yang
+```
+
+### 3. Define namespace as "http://github.com/Azure/{model-name}" and define a shorter(preferably within 3-4 letters) namespace prefix. Use the prefix in XPath for referring to any node present in other model.
+
+Example :
+####  YANG
+```
+module sonic-acl {
+	namespace "http://github.com/Azure/sonic-acl";
+	prefix sacl;
+	.....
+	.....
+}
+```
+
+### 4. Use 'revision' to record revision history whenever YANG model is changed. Updating revision with appropriate details helps tracking the changes in the model.
+
+Example :
+
+#### YANG
+```
+module sonic-acl {
+	revision 2019-09-02 {
+		description
+			"Added must expression for ACL_RULE_LIST."; // Updated with new change details
+	}
+	
+	revision 2019-09-01 {
+		description
+			"Initial revision.";
+	}
+
+	container sonic-acl {
+		.....
+		.....
+	}
+}
+```
+
+### 5. Each primary section of ABNF.json (i.e a dictionary in ABNF.json) for example, VLAN, VLAN_MEMBER, INTERFACE  in ABNF.json will be mapped to a container in YANG model.
+
+Example: Table VLAN will translate to container VLAN.
 
 #### ABNF
 
 ```
-"INTERFACE": {
-        "Ethernet112|2a04:fxxx:40:a709::xxx/126": {
-            "scope": "global",
-            "family": "IPv6"
+"VLAN": {
+        "Vlan100": {
+             "vlanid": "107"
         }
  }
 ```
 will translate to:
 
 ####  YANG
 --
+```
+container VLAN {  //"VLAN" mapped to a container
+ list VLAN_LIST {
+   key name;
+   leaf vlanid {
+      type uint16;
+   }
+ }
+}
+```
+
+### 6. By default table is defined in CONFIG_DB, if needed use extension 'sonic-ext:db-name' for defining the table present in other Redis DB.
```

Comment:
In that case, we can remove it from here. "sonic-ext:db-name" will be used when state data is being referred from COUNTER_DB/STATE_DB  etc, this can go in NB YANG specific steps.

Reply:
Agree, we remove it from here.